src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BedUsageReportGenerator.kt

```diff
@@ -26,7 +28,11 @@ class BedUsageReportGenerator(
 
   override val convert: BedEntity.(properties: BedUsageReportProperties) -> List<BedUsageReportRow> = { properties ->
     val startOfMonth = LocalDate.of(properties.year, properties.month, 1)
-    val endOfMonth = LocalDate.of(properties.year, properties.month, startOfMonth.month.length(startOfMonth.isLeapYear))
+    var endOfMonth = LocalDate.of(properties.year, properties.month, startOfMonth.month.length(startOfMonth.isLeapYear))
```

Comment:
could you please use a val here and assign using if/else to avoid value mutation

Reply:
We cant use if/else here because the endmonth has to referenced in DB query. The purpose of var is to change the value of the variable and i couldnt see any issue in using it in this context.

Reply:
In kotlin if statements can return values so you can do like this:

`
val endOfMonth = if (properties.serviceName == temporaryAccommodation && cas3EndDateOverride != 0) {
      startOfMonth.plusMonths(cas3EndDateOverride.toLong())
} else {
      LocalDate.of(properties.year, properties.month, startOfMonth.month.length(startOfMonth.isLeapYear))
}
`

I prefer to avoid vars because mutation typically makes code more complex. Obviously sometimes it's easier to just use a var as code as can get contrived otherwise but in this case i think the if/else construct works nicely. I'll approve regardless and up to you if you want to change it.

Reply:
nice, still learning kotlin :-). Done changes as suggetsed

Reply:
awesome, looks good!